std/assembly/dataview.ts

```diff
@@ -0,0 +1,131 @@
+import { HEADER_SIZE } from "./internal/arraybuffer";
+
+@inline
+function get<T>(buffer: ArrayBuffer, byteOffset: i32, littleEndian: boolean): T {
+  var result: T = load<T>(changetype<usize>(buffer) + byteOffset, HEADER_SIZE);
+  return littleEndian ? result : bswap<T>(result);
```

Comment:
Also `swap` not handle float points, so you need do this:

```ts
if (isFloat<T>()) {
  if (sizeof<T>() == 4) {
    let result = <u32>load<u32>(changetype<usize>(buffer) + byteOffset, HEADER_SIZE);
    return reinterpret<T>(littleEndian ? result : bswap<u32>(result));
  } else {
    let result = <u64>load<u64>(changetype<usize>(buffer) + byteOffset, HEADER_SIZE);
    return reinterpret<T>(littleEndian ? result : bswap<u64>(result));
  }
} else {
  let result: T = load<T>(changetype<usize>(buffer) + byteOffset, HEADER_SIZE);
  return littleEndian ? result : bswap<T>(result);
}
```

And for `set<T>` need to do similar